modules/core/core.js

```diff
@@ -148,6 +152,25 @@ exports.collections = async function(app) {
           format: 'checkbox',
           description: 'If set, collection will use a string id instead of PostgreSQL uuid'
         },
+        authRoutes: {
```

Comment:
@thomas4019 so this is one part i am not crazy about. Using the setting of these `authRoutes` to enable `auth` flow for the collection. It is not very intuitive, i would prefer something like a single boolean `requiresAuth`, and then automatically create all the routes based on a predefined pattern.

The problem is, is that the routes for `auth` are not actually that easy to infer without making breaking changes. For example `/user/login` is the singular, whereas the collection is plural `users`, same with `user/register/`. Which is why i went with a configured approach, allowing consumer to set them to whatever they want. 

However with `auth` i would prefer less flexibility to ensure security. Not sure if you have any ideas? One option is to introduce the breaking change where the auth routes are now simply `/:collection/login`, `/:collection/register`, `/:collection/me`, though i would still set them explicitly but just follow that pattern. 

Thoughts?

Reply:
Hmm, yeah, I agree this part is not very intuitive. I like the idea of a singe boolean `enableAuth` or maybe `enableLogin` and then just using `/:collection/login`, `/:collection/register`, `/:collection/me`. We can still hardcode the users collection to have the current ones.

So basically:
```javascript
if (collectionName === 'users') {
  return existing user routes 
} else {
  return simplified routes.
}
```

Reply:
Yeah thats a better solution. I would like us to move towards consistency for the routes so what would you think about allowing both sets of routes being created for users namely `user/login` and `users/login` with a deprecation warning for the old route? Something like:

```js
if (collectionName === 'users') {
  // add existing user routes 
  console.log('DEPRECATION: This route will be removed in future versions of Expressa, please switch to "....."')
}
// add simplified routes
```

Reply:
@thomas4019 i made some updates have a look and see what you think. I followed my idea above but without the warnings for now, im not overly attached to the idea so also happy to do as you suggested

Reply:
Added the deprecation warnings